[Feature] GET /boards/search 검색 기능 구현

windowM opened the pull request:
### 검색 기능 구현
- 현재는 title에 대해서만 검색을 허용함. (추후 기능 확장 예정)
- 게시글에 관련된 연관 엔티티들에 대해 fetch join을 걸어주었음
게시글과 wish를  `fetch join`으로 걸었다가 null이 나오는 문제가 있었음. `left fetch join`으로 수정 함.
(wish가 없더라도 결과는 나와야 하기 때문)

### 속도
- 속도 확인을 위해 시간 측정 코드를 남겨둠.
```
<조건>
- 데이터 수  
- 검색 조건  
- 첫 검색
```
 - N+1일때
 - Fetch Join
 - Elastic Search

windowM committed:
[feat] 게시글 상세 보기 기능 구현

windowM committed:
[feat] 게시글 삭제 기능 구현

windowM committed:
[feat] 게시글 검색 기능 구현

- fetch join 사용 member(inner), wish(left)

windowM committed:
[feat] db 변경 (h2->mysql)

windowM committed:
[docs] docker-compose.yml 파일 생성

Kong-TaeHyeon commented on src/main/java/com/example/fastboard/domain/board/controller/BoardController.java:
```diff
@@ -30,6 +31,14 @@ public ResponseEntity<ResponseDTO<Long>> create(
         return ResponseEntity.ok(ResponseDTO.okWithData(boardId));
     }
 
+    @GetMapping("/search")
```
* 페이지네이션에 대한 계획은 없으실까요? => 만약 있으시다면 어떤식으로 구현하실지 궁금합니다!

Kong-TaeHyeon commented on src/main/java/com/example/fastboard/domain/board/repository/BoardRepository.java:
```diff
@@ -2,9 +2,14 @@
+
     List<Board> findAllByDeletedAtIsNull();
+
+    @Query("SELECT b FROM Board b JOIN FETCH b.member LEFT JOIN FETCH b.wishes WHERE b.title LIKE %:title% ORDER BY b.createdAt DESC")
```
<img width="717" alt="image" src="https://github.com/user-attachments/assets/829c8b91-658f-414d-9b52-3c36a14025c4">

* 대부분 게시글 목록에는 댓글 수, 좋아요 수 등을 보여줄 것 같은데 이에 대한 처리는 어떻게 하시나요?
* 만약 처리하신다면, 단순히 LAZY 로 불러오는 방식으로 구현하시는걸까요?

Kong-TaeHyeon commented on src/main/java/com/example/fastboard/domain/board/service/BoardService.java:
```diff
@@ -48,9 +50,23 @@ public Long save(BoardCreateRequest request, Long memberId) {
+                .collect(Collectors.toList());
+
+        stopWatch.stop();
+        log.info("Execution time: " + stopWatch.getTotalTimeMillis() + " ms");
```
* 실행 시간에 대한 결과가 궁금해요! 
* 이를 테스트 코드에 작성하지 않고, 서비스 코드에 작성하신 이유가 있으실까요?

windowM committed:
[docs] sql쿼리 로그 정리하는 설정 추가

windowM committed:
[refactor] 2주차 리팩토링

<페이징 처리 & N+1 문제 처리>
- Entity 필드 List -> Set변환
  - 순서 보장이 안된다. comment를 불러올 때 문제가 발생 할 수 있을지도?
  - 중복데이터를 검사하는 로직이 set에 들어가기 때문에 성능상 느려질 수 도 있다고 한다.

- Fetch join -> Entity Graph
  - 페이징 처리를 할경우 이 작업은 메모리에서 수행된다. 그렇기 때문에 패치 조인을 사용하게 되면 모든 데이터를 들고온 후 메모리에서 페이징을 처리하기 때문에 성능 저하 및 문제 발생

※임시방편일 뿐 comment구현 후 다시 한번 해결해보자...

<이미지 경로 처리>

windowM committed:
[feat] 검색엔진 추가 (결과 출력만 확인)

<migrate>
- rdb board 데이터를 es 문서로 옮기는 api
- 추후 액션에 대한 업데이트를 처리할 수 있게 수정 할 예정.

<search>
- 현재 제목으로만 검색
- hit 점수를 기준으로 결과 정렬

windowM committed:
Merge pull request #45 from fast-board/feature/41-board-search-es

[feat] 검색엔진 추가 (결과 출력만 확인)

windowM committed:
Merge pull request #40 from fast-board/refactor/31-week2

[Refactor] 2주차 회의 리팩토링 진
